docs/tutorials/ruby-driver-create-client.txt

```diff
@@ -412,3 +412,63 @@ Details on Timeout Options
   interrupted on the server. See the
   `CRUD specification <https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst>`_ for details on
   operations that support this option.
+
+Details on Connection Pooling
+-----------------------------
+
+``Mongo::Client`` instances have a connection pool per server in your MongoDB topology. The pool opens connections on
+demand to support the number of concurrent MongoDB operations your application requires. There is no thread-affinity
+for connections.
+
+The client instance opens one additional connection per server in your MongoDB topology for monitoring the server's
+state.
+
+The size of each connection pool is capped at ``max_pool_size``, which defaults to 5. When a thread in your application
+begins an operation on MongoDB, it tries to retrieve a connection from the pool to send that operation on. If there
+are some connections available in the pool, it checks out a connection from the pool and uses it for the operation.
+If there are no connections available and the size of the pool is less than the ``max_pool_size``, a new connection
+will be created. If all connections are in use and the pool has reached its maximum, the thread pauses and waits for a
+connection to be returned to the pool by another thread.
+
+The number of seconds the thread will wait is configurable. This setting, called ``wait_queue_timeout``, is defined in
+seconds. It determines how long a thread will wait for a connection to be returned to the pool, when there are no
+connections available. If this timeout is reached, a ``Timeout::Error`` is raised. The default is 1 second.
+
+You can also set the number of connections initialized when the pool is created with the ``min_pool_size``
+setting. This is helpful if your application experiences load spikes and you want to avoid a slow-down when
+connections are created at the time of the spike. The default ``min_pool_size`` is 1.
+
+For example, a client connected to a 3-node replica set opens 3 monitoring sockets. It also opens as many sockets as
+needed to support a multi-threaded application's concurrent operations on each server, up to ``maxPoolSize``. If the
+application only uses the primary (the default), then only the primary connection pool grows and the total connections
+is at most 8 (5 connections for the primary pool + 3 monitoring connections). If the application uses a read
+preference to query the secondaries, their pools also grow and the total connections can reach 18 (5 + 5 + 5 + 3).
```

Comment:
Does the default min pool size of 1 apply to the secondary pools? I.e., does the driver open an additional connection proactively to each secondary, distinct from the monitors' connections?